packages/e2e-tests/specs/editor/various/rich-text.test.js

```diff
@@ -549,4 +551,81 @@ describe( 'RichText', () => {
 		// Expect: <strong>1</strong>-<em>2</em>
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	// Regression test for https://github.com/WordPress/gutenberg/pull/35516.
+	it( 'should properly convert `text-color` <span> tags to <mark> tags', async () => {
+		// Change editing mode from "Visual" to "HTML".
+		await switchEditorModeTo( 'Code' );
+
+		const markupFixture = `
+				<!-- wp:paragraph -->
+				<p><span class="has-inline-color has-red-color">Preset color added pre-11.7</span></p>
+				<!-- /wp:paragraph -->
+
+				<!-- wp:paragraph -->
+				<p><span style="color:#a30097" class="has-inline-color">Custom color added pre-11.7</span></p>
+				<!-- /wp:paragraph -->
+			`;
+
+		await page.type( '.editor-post-text-editor', markupFixture );
+
+		// Change editing mode from "HTML" to "Visual"
+		await switchEditorModeTo( 'Visual' );
+
+		// Confirm editor markup has been properly converted
+		const [
+			editorPresetColorText,
+			editorCustomColorText,
+		] = await page.evaluate( () =>
+			Array.from(
+				document.querySelectorAll(
+					'p[aria-label="Paragraph block"] > *'
+				)
+			).map( ( el ) => ( {
+				tag: el.localName,
+				style: el.getAttribute( 'style' ),
+			} ) )
+		);
+
+		expect( editorPresetColorText.tag ).toEqual( 'mark' );
+		expect( editorPresetColorText.style ).toEqual(
+			'background-color:rgba(0, 0, 0, 0)'
+		);
+		expect( editorCustomColorText.tag ).toEqual( 'mark' );
+		expect( editorCustomColorText.style ).toEqual(
+			'background-color:rgba(0, 0, 0, 0);color:#a30097'
+		);
+
+		// Publish and view post
+		await publishPost();
+		const viewPostLinks = await page.$x(
+			"//a[contains(text(), 'View Post')]"
+		);
+		await viewPostLinks[ 0 ].click();
+		await page.waitForNavigation();
```

Comment:
We should be wrapping this with `Promise.all` to be sure we don't start listening after the navigation event occured:

```suggestion
		await Promise.all( [
			page.waitForNavigation(),
			viewPostLinks[ 0 ].click()
		] );
```

Reply:
Yep! The [advised way](https://pptr.dev/#?product=Puppeteer&version=v12.0.1&show=api-pagewaitfornavigationoptions) is to start waiting for the navigation event **before** the click occurs. This way we're sure that we don't accidentally start waiting after the navigation event (triggered by the click) finished, and end up with the "Timed out waiting for navigation" error. Does that make sense?

Reply:
Ah... if you start waiting for it before it even begins, you know you won't miss it. Yeah, that makes sense, thanks!